### PR TITLE
[Win] Part 2: Make frontend plugin Windows compatible.

### PR DESCRIPTION
**Pull request by fodinabor**
This reverts to always using Itanium mangling, but on Windows anyways relies on this [LLVM patch](https://reviews.llvm.org/D69322) to get consistent lambda numbering.
Thus, the new API, introduced in that patch, is always used on Windows, to get the correct behavior.

Also included is a hack to fix some compiler crashes, caused by nullptrs being handed into `TraverseDecl`.

### CODE REVIEW

File: include/hipSYCL/compiler/Frontend.hpp [hunk 1]
```diff
@@ -167,8 +172,14 @@ class FrontendASTVisitor : public clang::RecursiveASTVisitor<FrontendASTVisitor>
 public:
   FrontendASTVisitor(clang::CompilerInstance &instance)
       : Instance{instance}
-      , KernelNameMangler(instance.getASTContext().createMangleContext())
-  {}
+      , KernelNameMangler(clang::ItaniumMangleContext::create(instance.getASTContext(), instance.getASTContext().getDiagnostics()))
```
Comment (illuhad) [inline include/hipSYCL/compiler/Frontend.hpp]: Maybe add a line break?

**Comment by illuhad** [comment]
@hipSYCL-ci-bot test with various clangs :)

**Review by illuhad — APPROVED** [review]

**Comment by psalz** [comment]
So Clang plugins now work on Windows huh? Interesting... 🤔

**Comment by fodinabor** [comment]
It's possible with clang.exe (not clang++ or clang-cl).. But as *all* symbols are exported, it might happen at some point that there are too many (don't remember the limit, but still a bit to go) and everything will break.. So won't be officially supported at any point in time I guess...

**Pull request merged**